pkg/daemon/daemon.go

```diff
@@ -149,19 +157,128 @@ func (d *daemon) GetIbSriovNetwork(networkID string) (string, *utils.IbSriovCniS
 	return networkName, ibCniSpec, nil
 }
 
+// Get Pod's IB network specified by name. Update netMap with Pod's networks.
+func getIbSriovPodNetwork(netName string, pw *PodWrapper, netMap map[types.UID][]*v1.NetworkSelectionElement) error {
```

Comment:
in this method its not clear what is the input and what is the output.
also not too keen on the PodWrapper where it being used for both input and output

does it make sense just to update the pod network map here ? (and rename the method ofc)
main flow can just use : utils.GetPodNetwork()

Reply:
> IMHO we need to move all the logic of the cache network to different class
> NetworkCache class

What do You mean under "logic of the cache network"? If map update, then "all logic" is currently this map update only.

> get(netName,pw *PodWrapper)
> if netMap[pod id] exist return the NetworkSelectionElement
> else extract it from the PodWrapper.

Wrapper is temporary variable in loop, it doesn't hold networks from previous main loop iteration.

Reply:
so I want to warp netMap map[types.UID] the in a class/struct and move the logic of checking the map or getting i from PodWrapper in it so it won't be part of the daemon struct

Reply:
Here's what I got for this:
```
type networksMap struct {
        theMap map[types.UID][]*v1.NetworkSelectionElement
}

// Return networks mapped to the pod. If mapping not exist it is created
func (n *networksMap) getPodNetworks(pod *kapi.Pod) ([]*v1.NetworkSelectionElement, error) {
        var err error
        networks, ok := n.theMap[pod.UID]
        if !ok {
                networks, err = netAttUtils.ParsePodNetworkAnnotation(pod)
                if err != nil {
                        return nil, fmt.Errorf("failed to read pod networkName annotations pod namespace %s name %s, with error: %v",
                                pod.Namespace, pod.Name, err)
                }

                n.theMap[pod.UID] = networks
        }
        return networks, nil
}
```
And `getIbSriovPodNetwork` function changed to `getPodNetworkInfo` which return `podInfo` (exWrapper), but without GUID as was initially.